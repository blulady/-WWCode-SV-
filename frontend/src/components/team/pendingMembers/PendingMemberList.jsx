import React from "react";
import styles from "./PendingMemberList.module.css";
import "../../../Common.css";

const PendingMemberList = (props) => {
    const users = props.users;

    return (
        
        users.map((user, idx) =>
            <div className={styles["pending-member-list-card"] + " d-flex flex-column"} key={idx}>
                <div className={"align-self-end " + styles["icon"] + " " + styles["more"]}></div>
                <div className={"d-flex " + styles["row"]}>
                    <div className={styles["column"]}>Email</div>
                    <div className={styles["column"]}>{user.email}</div>
                </div>
                <div className={"d-flex " + styles["row"]}>
                    <div className={styles["column"]}>Role</div>
                    <div className={styles["column"] + " wwc-text-capitalize"}>{user.role_name.toLowerCase()}</div>
                </div>
                <div className={"d-flex " + styles["row"]}>
                    <div className={styles["column"]}>Status</div>
                    <div className={styles["column"] + " wwc-text-capitalize"}>{user.status.toLowerCase()}</div>
                </div>
                <div className={"d-flex " + styles["row"]}>
                    <div className={styles["column"]}>Resend Invite</div>
                    <div className={styles["column"]}><button className={styles["invite-button"] + " " + styles["icon"]} data-bs-toggle="modal" data-bs-target={props.target} data-bs-user={user.email}>Resend Invite</button></div>
                </div>
            </div>
        )
    );
}

export default PendingMemberList;