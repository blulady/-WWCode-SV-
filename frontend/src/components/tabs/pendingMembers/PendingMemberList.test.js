import React from "react";
import { render } from "@testing-library/react";
import PendingMemberList from "./PendingMemberList";

const pendingMembers = [{
    email: "abc@example.com",
    role_name: "volunteer",
    status: "invited"
}];

describe("PendingMemberList Component Validation Tests", () => {

  test("PendingMemberList component is rendering", async () => {
    const { container, getByText } = render(<PendingMemberList users={pendingMembers} />);
    expect(container).toMatchSnapshot();

    expect(getByText("abc@example.com")).toBeTruthy();
    expect(getByText("volunteer")).toBeTruthy();
    expect(getByText("invited")).toBeTruthy();
  });
});
