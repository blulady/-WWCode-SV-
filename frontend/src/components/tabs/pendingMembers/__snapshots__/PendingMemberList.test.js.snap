// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`PendingMemberList Component Validation Tests PendingMemberList component is rendering 1`] = `
<div>
  <div
    class="pending-member-list-card d-flex flex-column"
  >
    <div
      class="align-self-end"
    >
      <div
        aria-expanded="false"
        class="icon more"
        data-bs-toggle="dropdown"
      />
      <ul
        class="dropdown-menu  dropdown-menu-lg-start"
      >
        <li>
          <button
            class="dropdown-item"
            data-bs-target="#deletePendingMemberDialog-undefined"
            data-bs-toggle="modal"
          >
            Delete
          </button>
        </li>
      </ul>
    </div>
    <div
      class="d-flex row"
    >
      <div
        class="column"
      >
        Email
      </div>
      <div
        class="column"
      >
        abc@example.com
      </div>
    </div>
    <div
      class="d-flex row"
    >
      <div
        class="column"
      >
        Role
      </div>
      <div
        class="column wwc-text-capitalize"
      >
        volunteer
      </div>
    </div>
    <div
      class="d-flex row"
    >
      <div
        class="column"
      >
        Status
      </div>
      <div
        class="column wwc-text-capitalize"
      >
        invited
      </div>
    </div>
    <div
      class="d-flex row"
    >
      <div
        class="column"
      >
        Resend Invite
      </div>
      <div
        class="column"
      >
        <button
          class="invite-button icon"
          data-bs-target="#resendConfirmationDialog-undefined"
          data-bs-toggle="modal"
          type="button"
        >
          Resend Invite
        </button>
      </div>
    </div>
  </div>
</div>
`;
